### ISSUE TITLE
Can't create const array of named enums from autoload singleton

### ISSUE THREAD

**Issue by KoBeWi** [godotengine/godot#52832]
### Godot version

b1063a7

### System information

W10

### Issue description

So this code
```
## Autoload.gd
enum Test {TEST1, TEST2}
```
```
## Whatever.gd
const a = [Autoload.Test.TEST1, Autoload.Test.TEST2]
```

Doesn't work. It was fine in 3.x
It works if the enum is unnamed, e.g.
```
const a = [Autoload.TEST1, Autoload.TEST2]
```

I somehow got 2 different errors. There was one saying that it expects a constant expression, but in a minimal project the error says that can't create Variant array of unresolved type.

### Steps to reproduce

1. Add autoload and script with above code

**Comment by haskjold** [comment]
I'm also having problems with named enums and autoload, but with some different errors.

My console gets filled up with messages like these:
```
ERROR: Native class res://director.gd.Scenes used in script doesn't exist or isn't exposed.
   at: get_function_signature (modules/gdscript/gdscript_analyzer.cpp:3601)
ERROR: Native class res://director.gd.Scenes used in script doesn't exist or isn't exposed.
   at: get_function_signature (modules/gdscript/gdscript_analyzer.cpp:3601)
...
```

Where Scenes is a named enum.

I'm also getting crashes when trying to use named enums in autoloads:

```
================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.0.dev.20211117.official (1756909577fc1758632153a97668fe9843031a32)
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] /lib/x86_64-linux-gnu/libc.so.6(+0x46210) [0x7f765edf6210] (??:0)
-- END OF BACKTRACE --
================================================================
```

**Comment by Zylann** [comment]
Also having something similar, but not autoloads, instead with a preloaded script. Something like:
```gdscript
const Foo = preload("foo.gd")
const A = [[Foo.SOME_CONSTANT]]
```
Would this need a different issue due to using a preload instead of autoload?

**Comment by dalexeev** [comment]
> ```gdscript
> const A = [[Foo.SOME_CONSTANT]] # observe the nesting
> ```

#50285

**Comment by KoBeWi** [comment]
Seems fixed in a12fe31
